examples/multi-engine/index.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+/**
+ * Module dependencies.
+ */
+
+var express = require('../../');
+var ejs = require('ejs');
+var hbs = require('hbs');
+var path = require('path');
+var fs = require('fs');
+
+var viewsPath = path.join(__dirname, 'views');
+
+/**
+ * @description Helper function to render template string
+ * from one template engine inside another one.
+ *      Note:
+ *          for demo purpose only one template engine (HBS)
+ *          is added to the strategy, but you can add as many
+ *          as you want to. Optimization on HBS can be applied
+ *          via precompiling or on demand caching.
+ * @example
+ *        EJS template
+ *           - render(HBS template)
+ */
+function render(relativePath, options) {
+  var absolutePath = path.join(viewsPath, relativePath);
+  var engine = relativePath.split('.').pop();
+
+  var strategy = {
+    'hbs': function() {
+      var str = fs.readFileSync(absolutePath, 'utf8');
+      var template = hbs.handlebars.compile(str);
+      return template(options);
+    },
+  }
+
+  return (strategy[engine] && strategy[engine]())
+    || '<!-- unsupported engine -->';
+}
+
+var app = express();
+
+app.engine('html', ejs.renderFile);
+app.engine('ejs', ejs.renderFile);
+app.engine('hbs', hbs.__express);
+
+app.set('view engine', 'html');
+app.set('views', viewsPath)
+
+/**
+ * @description entrypoint for demo
+ * default ejs template will include 1 EJS template and will
+ * try to render 2 others, namely HBS and PUG. Since we
+ * have not configured PUG in our strategy above, it will
+ * output a comment.
+ *    Input (psudo):
+ *        EJS template
+ *          - include EJS template
+ *          - render HBS template
+ *          - render PUG template
+ *    Output:
+ *        <div>template_1: EJS</div>
+ *        <div> template_2: HBS, rendered inside EJS</div>
+ *        <!-- unsupported engine -->
+ *    Note:
+ *        we need to pass a callback function (called inhere
```

Comment:
would it make sense to do `app.locals.render = render` at the top level so it's always available instead of needing to pass it in?

Reply:
ah, yes absolutely! Thanks for point this one out. New express user here, so still learning the interface :) I have updated the PR.